cpp/src/io/parquet/writer_impl.cu

```diff
@@ -232,39 +378,113 @@ class parquet_column_view {
+      auto d_col = column_device_view::create(col, stream);
+      get_leaf_offset<<<1, 1, 0, stream>>>(*d_col, d_col_offset.data(), d_end_offset.data());
+      leaf_col_offset = d_col_offset.value(stream);
+      _data_count     = d_end_offset.value(stream) - leaf_col_offset;
+
+      _type_width = (is_fixed_width(_leaf_col.type())) ? cudf::size_of(_leaf_col.type()) : 0;
+      _data       = (is_fixed_width(_leaf_col.type()))
+                ? _leaf_col.head<uint8_t>() + leaf_col_offset * _type_width
+                : nullptr;
+
+      // Bring offset array to device
+      column_view curr_col = col;
+      std::vector<size_type const *> offsets_array;
+      while (curr_col.type().id() == type_id::LIST) {
+        lists_column_view list_col(curr_col);
+        offsets_array.push_back(list_col.offsets().data<size_type>());
+        curr_col = list_col.child();
+      }
+      _offsets_array = offsets_array;
+
+      // Calculate row offset into dremel data (repetition/definition values)
+      _dremel_offsets.resize(_row_count + 1);
+      dremel_encoding_size<<<((_row_count - 1) >> 8) + 1, 256, 0, stream>>>(
+        *d_col, _dremel_offsets.data().get());
+      CUDA_TRY(cudaStreamSynchronize(stream));
+
+      thrust::exclusive_scan(rmm::exec_policy(stream)->on(stream),
+                             _dremel_offsets.begin(),
+                             _dremel_offsets.end(),
+                             _dremel_offsets.begin());
+      CUDA_TRY(cudaStreamSynchronize(stream));
+      CHECK_CUDA(stream);
+
+      // Calculate definition and repetition levels
+      size_type flattened_size = _dremel_offsets.back();
+      _rep_level.resize(flattened_size);
+      _def_level.resize(flattened_size);
+      CUDA_TRY(cudaDeviceSynchronize());
+      calculate_levels<<<((_row_count - 1) >> 8) + 1, 256, 0, stream>>>(
+        *d_col, _dremel_offsets.data().get(), _rep_level.data().get(), _def_level.data().get());
+    }
     if (_string_type && _data_count > 0) {
-      strings_column_view view{col};
+      strings_column_view view{_leaf_col};
       _indexes = rmm::device_buffer(_data_count * sizeof(gpu::nvstrdesc_s), stream);
       stringdata_to_nvstrdesc<<<((_data_count - 1) >> 8) + 1, 256, 0, stream>>>(
         reinterpret_cast<gpu::nvstrdesc_s *>(_indexes.data()),
-        view.offsets().data<size_type>() + view.offset(),
+        view.offsets().data<size_type>() + leaf_col_offset,
         view.chars().data<char>(),
         _nulls,
         _data_count);
       _data = _indexes.data();
       CUDA_TRY(cudaStreamSynchronize(stream));
     }
+
     // Generating default name if name isn't present in metadata
     if (metadata && _id < metadata->column_names.size()) {
       _name = metadata->column_names[_id];
     } else {
       _name = "_col" + std::to_string(_id);
     }
+    _path_in_schema.push_back(_name);
   }
 
   auto is_string() const noexcept { return _string_type; }
+  auto is_list() const noexcept { return _list_type; }
```

Comment:
```suggestion
  auto is_nested() const noexcept { return _nested_type; }
```

Reply:
As of right now, I'm not making any guarantee this will work with other nested types. I'm planning to keep it as list type until struct support.

Reply:
Sounds good.